### ISSUE TITLE
[TF2] [Feature Request] Increase maximum fov_desired value

### ISSUE THREAD

**Issue by AndrewBetson** [ValveSoftware/Source-1-Games#4043]
# Issue
While the current maximum FOV of 90 is fine for most players, some (including myself) prefer to play FPS games with an FOV in the range of 100-120; having a higher FOV doesn't give players any meaningful advantage over others, and forcing the FOV higher than 90 with server plugins doesn't cause any significant visual errors.

# Proposal
Increase the maximum value of the `fov_desired` cvar to somewhere between 100-120.

**Comment by condor00fr** [comment]
the default fov should also be 90, as used by the majority of people on most fps games theses days, and the default weapon fov should be 70 for the same reasons

**Comment by Ashetf2** [comment]
Valve locked FOV in CS:GO to 90 for 16:9 res, so i guess there's a reason they don't want people to have higher FOV values

**Comment by xcloudx01** [comment]
FOV looks kinda odd in 32:9 super ultra-wide, I wish i could increase it just a tad higher, like 95 or something

**Comment by tmob03** [comment]
Yes, please add this. 120 fov was added in the recent updates to hl2 (episodes), why not tf2? If you can't be bothered to fix the view clipping through walls, limit fov to like 105, I believe it doesn't happen much or at all at that fov. 

People might say this gives players an advantage over others, sure. Yet this was completely okay going all the way back to quake days. If this is really a concern, just force 90 fov in competitive environments, simple as that.

**Comment by Radial36** [comment]
As a more-or-less major update is on the horizon, I wanted to bump this so it gets seen and hopefully implemented. Its a great QOL improvement for with no major downsides. Maybe clamp to 90 in competitive matches? There is no real need to fix camera clipping through walls, as the game already allows peeking around corners via taunts, so wall clipping would be a non-issue balance-wise.

Furthermore, L4D2 received this same change some time ago and, despite being a significantly more competitive-oriented game than TF2, the wall clipping that results from high FOV has never been an issue balance-wise, and has actually only been a major improvement, for the obvious reason of just seeing more of the game on screen.

**Comment by Ashetf2** [comment]
> Yes, please add this. 120 fov was added in the recent updates to hl2 (episodes), why not tf2? If you can't be bothered to fix the view clipping through walls, limit fov to like 105, I believe it doesn't happen much or at all at that fov.
> 
> People might say this gives players an advantage over others, sure. Yet this was completely okay going all the way back to quake days. If this is really a concern, just force 90 fov in competitive environments, simple as that.

HL2 is a single player game.

Again, Valve locked FOV to 90 in CS:GO for 16:9 for some reason.

**Comment by Radial36** [comment]
> Valve locked FOV in CS:GO to 90 for 16:9 res, so i guess there's a reason they don't want people to have higher FOV values

CSGO is a completely different game with a competitive scene that valve actively supports. Why are you comparing the two, and saying a change not being done in CSGO means the same should be done for TF2?

Allowing FOV up to 120 is completely fine, and will not cause any issues with game balance. If you are worried about competitive, community and valve comp can simply limit FOV to 90.

**Comment by krizej** [comment]
> There is no real need to fix camera clipping through walls, as the game already allows peeking around corners via taunts, so wall clipping would be a non-issue balance-wise.

Not only that, but the camera clipping exploit would not be practical in most, if not all, cases. It reveals only a tiny bit, and most likely it will be too awkward to adjust the angle and see enemy players/buildings/whatever. I'm not even going to mention TF2's pacing.

**Comment by LiquifiedPizzas** [comment]
Fixing the seeing through walls thing could most likely be done by a small adjustment to how close the Near Z clipping plane is to the camera's position. Either way, I think the game supporting ultrawide (allowing people with 32:9 monitors to have a horizontal FOV of around 140) is pretty absurd to have if your justification for locking the FOV to 90 is some perceived balance reason. Either you disallow ultrawide players seeing more or raise the FOV cap for everyone, and I think the latter is preferable for freedom of choice. Additionally, many pros still play on 4:3 stretched to 16:9, so I don't think the FOV makes as big of a deal to balance as some people think anyways: it would just be good quality of life to have. An incentive for people to buy an ultrawide monitor just to have an FOV they like in an FPS game is a little odd.

**Comment by kylejrp** [comment]
This only applies to windowed mode, right? Valve servers have `sv_restrict_aspect_ratio` set to `1` which restricts the FOV in windowed mode but not full screen mode. If you have an ultrawide, you just have to play in full screen mode?

**Comment by LiquifiedPizzas** [comment]
> This only applies to windowed mode, right? Valve servers have `sv_restrict_aspect_ratio` set to `1` which restricts the FOV in windowed mode but not full screen mode. If you have an ultrawide, you just have to play in full screen mode?

Yeah, if you play in fullscreen the ultrawide FOV works as intended, and to be honest, I wouldn't want that changed. I just want people on other aspect ratios to be able to not be locked out of having FOVs like that.

**Comment by pitbullnamed1** [comment]
As a heavy main and a former scout main, I can say with 100% certainty that the reason why spy hard counters heavy is because you can't see anything with 90 fov. Overriding your resolution to a 24:9 ratio provides a rather displeasing tf2 experience, but at least classes that require situational awareness suddenly have a valid way to keep being in fact, aware of their surroundings, like spies, blast jumping enemies, and good scouts? It's horrifying how many players just live with the fact that they will never counter spies and players 10 meters above them ever unless they are wall hacking, and that is not legal. Capping fov_desired to 120 is the best thing Valve could do to tf2, I don't care about bots nor cheaters, both are curbed by either just playing better or requeueing.

**Comment by wgetJane** [comment]
@pitbullnamed1 heavy counters spy (the minigun is the best spychecking weapon)

**Comment by tompa71a** [comment]
Would love to see an increase of the max fov. In most other fps games I play with fov 100-120. It just feels better and smoother. More fun to play.

**Comment by A1steaksa** [comment]
A higher max FOV would substantially increase my experience playing the game.

An FOV limit of 90° can be—and in my case is—somewhat uncomfortable to play with.

**Comment by webbongithub** [comment]
It's important to note this isn't just a nice-to-have, it's a crucial accessibility feature. I have simply been unable to play vanilla TF2 for some years now because I can't increase it past ninety. It is literally painful and nauseating to play at 90. There's no reason that at least in casual contexts that the maximum FOV shouldn't be much higher. Personally, I think the accessibility wins far outweighs any gameplay changes -- which the meta will adjust to in the case of online multiplayer games. I'd honestly take the option of having a higher FOV but having entities only render in the area on my screen that would corrospond to 90FOV if it was that or nothing.

**Comment by movercell** [comment]
This^ i'm just completely unable to play the game without getting motion sickness

**Comment by blackletum** [comment]
Since my [PR](https://github.com/ValveSoftware/source-sdk-2013/pull/994/) was closed, pasting my same thoughts here:

This would help players on standard widescreen monitors be able to see as much as players on ultrawide displays, help reduce nausea, to have parity with Valve's other titles such as Left 4 Dead 2 as well as the Team Fortress modding communities typically allowed max FOV settings.

**Comment by Sirbucket** [comment]
I would rather a dedicated ultrawide view option is made for ultrawide monitors instead of raising the fov and making the game look like garbage for greater vision

**Comment by Sirbucket** [comment]
If it is to be increased I would go with 100 as gmod has it and is a prime example of 100 fov being perfectly tolerable in source games.

**Comment by Lambdagon** [comment]
making the game "look like garbage for greater vision" sounds like your personal preference. You aren't the only tf2 player and you are not FORCED to use 130 fov. There are many people that use graphics configs that make their game look terrible and still enjoy it. I don't see how garbage looks should affect this change. Multiple source mods already have the cap increased to 130

**Comment by Menoly12** [comment]
There was also a comment from that closed PR about it forcing other players to play the same way due to competitive advantages, when one of the top players play in stretched 4:3, which would be lower FOV than what typical widescreen players use.

**Comment by Margen67** [comment]
> I would rather a dedicated ultrawide view option is made for ultrawide monitors instead of raising the fov and making the game look like garbage for greater vision

@Sirbucket You mean cropping the view like [Vert-](https://www.wsgf.org/book/export/html/908)?

**Comment by webbongithub** [comment]
> I would rather a dedicated ultrawide view option is made for ultrawide monitors instead of raising the fov and making the game look like garbage for greater vision

There is no actual evidence that, in the case of TF2, raising the FOV significantly affects gameplay and visibility, whereas there is a lot of evidence that it makes the game painful/nauseating for people. Even if it did I would posit it would be small enough of a difference that the accessibility win outweighs any gameplay advantage.

The heavily distorted edges are, in practice, basically useless for seeing much beyond "there's things happening in my peripheral". You also get a nerf in the centre of your vision, as things are much smaller. In many ways visibility is *decreased*, and overall I would say it's lower. 

Having an ultrawide view option (I think what you mean is emulating an ultrawide monitor and using black bars) would not solve the problem for me as it wouldn't decrease apparent motion, which increasing the maximum FOV does. It also makes the game extremely hard to see. I need all the pixels on my monitor.

I would encourage people in this thread to try TF2 Classic where the maximum is 130 and see how it affects the game. The difference in FOV is basically nothing when moving your head is an extremely easy and cheap thing to do. As someone who plays on 125 FOV, I can tell you that I get backstabbed just as much if not more than before, and Spy is not ruined by it (hell, it's better there because Spy is less buggy.)

**Comment by RobinsAviary** [comment]
Just throwing in my two cents here - I find that a max of like 100-110 to be reasonable in modern FPS games. I mean, I play minecraft at 92 FOV, which even that is technically too high for tf2. I think it would be nice to raise by 10-20 degrees, and I really don't think it would affect gameplay that much, speaking from personal experience.

**Comment by blackletum** [comment]
> Just throwing in my two cents here - I find that a max of like 100-110 to be reasonable in modern FPS games. I mean, I play minecraft at 92 FOV, which even that is technically too high for tf2. I think it would be nice to raise by 10-20 degrees, and I really don't think it would affect gameplay that much, speaking from personal experience.

92 fov gang (idk why I like 92 but it's what I use in most games)

But yes I'm all for players being able to use the FOV that they like best in game. Many find that they have issues with nausea with lower FOVs too, and I can say after using 110 in Pre-Fortress 2 for the longest time, even I find the lower FOV in TF2 "uncomfortable" to play with.

**Comment by movercell** [comment]
This.

It also doesn't affect the competitive leagues since they can enforce 90 FOV by themselves so there's nothing stopping the devs from removing low-fov-induced motion sickness

**Comment by Azkuzku** [comment]
Another approach to this could be having the game render in a way similiar to GoldSrc, where if you play at a resolution like 1280x1024/1680x1050 instead of cropping the sides it uncrops the top and bottom thus, giving you a higher fov more akin to 16:9 in 4:3. This would of course have to be limited to some degree, otherwise vertical resolutions would become exploitable.

**Comment by HarmlessAphorism** [comment]
> Another approach to this could be having the game render in a way similiar to GoldSrc, where if you play at a resolution like 1280x1024/1680x1050 instead of cropping the sides it uncrops the top and bottom thus, giving you a higher fov more akin to 16:9 in 4:3. This would of course have to be limited to some degree, otherwise vertical resolutions would become exploitable.

I agree with you that switching to Vert+ for resolutions narrower than 16:9 would be a good thing. It would make sure everyone can at least see as much as 16:9. The best games with arbitrary aspect ratio support tend to do this, I've noticed. I will say, regarding exploitability though, the current HOR+ scaling in the game is already quite exploitable:

<img width="2560" height="720" alt="Image" src="https://github.com/user-attachments/assets/1d3cb810-f8ff-40cc-9343-4842e22500ba" />

Screenshot on an official valve server. Unsure why tall resolutions would deserve that treatment unless both tall and wide resolutions both get a cap on one axis somewhere. Additionally, this game is not usually played for maximum competitive integrity outside of league: you can make a new server variable that clamps the max FOV allowed for leagues to tinker with, while leaving it more lenient in defaults and casual and such.

More generally about this topic:

It is not hard to load up CRU, make a custom resolution (this one is 32:9) and get a horizontal FOV of 140. This has been available for forever now, and I don't think it's exactly shattered the meta or anything. I agree that maybe there should be a limit at some point, but I don't even think TF2C's 130 limit noticeably changes gameplay dynamics. In the normal game, top tier players are not going out of their way to play in superultrawide or buy expensive 21:9 and 32:9 monitors: intsead, some of the best people in the world intentionally hinder their FOV with stretched 4:3 and do just fine. This applies to CS as well, for another example. I don't think it makes as big of a difference as people think. As webbong puts earlier, there is almost no actual convincing evidence given as to the negative effect of increasing FOVs, it is almost all conjecture and edge cases, with very little in the way of concrete results. Meanwhile, there is plenty of evidence of the negative effect of low FOVs on people, even if a lot is anecdotal.

Lastly, a part of PC gaming is customizability and accessibility. Valve, a PC juggernaut, should understand this. People in this game have the freedom to make this game look like absolute trash, with basically no textures at all, delete models from the game files and still play casual, even edit particle stuff that genuinely gives you an advantage. Being able to officially raise the FoV should not be a bridge too far. HL2 and Portal got FoV updates somewhat recently and it's 2025, TF2 players deserve to be able to select higher than 74 degrees vertical.

Anecdotally, I have a friend that loved TF2 in the past, but tells me he cannot play it anymore, as the low FOV makes him uncomfortable after 20 or so minutes. It's frustrating to have to deal with this, I would greatly appreciate it being increased. Every aspect ratio should be able to get to an FoV they feel is comfortable (within reason), as messing with aspect ratio just to get FoVs you want is just silly.

**Comment by a584-droid** [comment]
Fov 100 looks quite good; should be an option between 90-100.

**Comment by webbongithub** [comment]
100 is too low to be playable for me and likely others.

**Comment by movercell** [comment]
Yeah, need at least 110

**Comment by ghost** [comment]
Giving players more control over theirs gaming experience has been the norm since the very begining and there's no reasons why TF2 shouldn't offer more, easily accessible options for anyone to play more comfortably.

Wolfenstein 3D, for instance, had built in support for mouse and keyboard even through only a a few could play like this in 92.

I don't see why allowing players to change theirs fov would remotely affect this game in a negative way anyways, the game's casual nature is what most appeals to most people in the first place so it's not like it'll cause problems.

Besides the first thing any friendly players teach you after you tell the them that you are new is to change your fov to at least 90 and your weapon fov to at least 70 or whatever else you are comfortable with.

Lmao fov 90 and weapon fov 70 should be the default settings in the first place with the option to change your fov to a higher status should you want to be available.

**Comment by webbongithub** [comment]
I should mention that the general consensus amongst mods that choose to increase it (Open Fortress, Team Fortress 2 Classified, Pre-Fortress) is to set the maximum to 130. I have never once heard complaints in the community about how high it goes and I think that's a good maximum.

For me I could probably push my FOV lower, but 130 has always been a "safe" value in that I never need to adjust it and I never experience problems. Sometimes I'm more sensitive, sometimes I'm not, but I'm always good with it.

Vanilla TF2 being so low is the exception, and bringing it to 130 would bring it in line with most mods of the game.

**Comment by GreenSwede** [comment]
My opinion: Default fov should increase and remain at 90. 75 fov made sense back when 1920x1080 wasn't the norm, many players were still using _1024x768 4:3 aspect ratio_ when TF2 first came out back in 2007.

I don't think allowing high values (anything above 100) is fair because you'll put players with a lower fov at an obvious disadvantage.
